app/src/common/shared/com/igalia/wolvic/ui/widgets/dialogs/LegalDocumentDialogWidget.java

```diff
@@ -109,7 +109,13 @@ protected void initializeWidgetPlacement(WidgetPlacement aPlacement) {
         aPlacement.anchorY = 0.5f;
         aPlacement.translationY = WidgetPlacement.unitFromMeters(getContext(), R.dimen.settings_world_y) -
                 WidgetPlacement.unitFromMeters(getContext(), R.dimen.window_world_y);
-        aPlacement.translationZ = WidgetPlacement.unitFromMeters(getContext(), R.dimen.settings_world_z) -
-                WidgetPlacement.unitFromMeters(getContext(), R.dimen.window_world_z);
+        updatePlacementTranslationZ();
+    }
+
+    @Override
+    public void updatePlacementTranslationZ() {
+        getPlacement().translationZ = WidgetPlacement.unitFromMeters(getContext(), R.dimen.settings_world_z) -
+                WidgetPlacement.getWindowWorldZMeters(getContext());
+        updateWidget();
```

Comment:
This adds a new behaviour in some callsites that were not calling updateWidget(). Can we move updateWidget() to the callers and leave these methods with just the translationZ assignment?

Reply:
Updated